lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js

```diff
@@ -21,13 +22,63 @@ module.exports = {
             const errorMessage = [
               `HTTP event of function ${functionName} is not an object nor a string.`,
               ' The correct syntax is: http: get users/list',
-              ' OR an object with "path" and "method" proeprties.',
+              ' OR an object with "path" and "method" properties.',
               ' Please check the docs for more info.',
             ].join('');
             throw new this.serverless.classes
               .Error(errorMessage);
           }
 
+          // Setup cors.
+          let cors;
+          let corsEnabled = false;
+          if (!!event.http.cors) {
+            corsEnabled = true;
+            const headers = [
+              'Content-Type',
+              'X-Amz-Date',
+              'Authorization',
+              'X-Api-Key',
+              'X-Amz-Security-Token'];
+
+            cors = {
+              origins: ['*'],
+              methods: ['OPTIONS'],
+              headers,
+            };
+
+            if (typeof event.http.cors === 'object') {
+              cors = event.http.cors;
+              cors.methods = [];
+              if (cors.headers && !Array.isArray(cors.headers)) {
+                const errorMessage = [
+                  'CORS header values must be provided as an array.',
+                  ' Please check the docs for more info.',
+                ].join('');
+                throw new this.serverless.classes
+                .Error(errorMessage);
+              } else {
+                cors.headers = headers;
```

Comment:
I should have been more specific 😅  I was referring to L61. I think it was overwriting the headers in set my http event definition. I opened #1984 to track.